src/snakemake/jobs.py

```
@@ -64,34 +64,43 @@
 
 def format_file(f, is_input: bool):
+    if f.is_storage:
+        if is_input:
+            if f.storage_object.retrieve:
+                storage_phrase = "retrieve from"
+            else:
+                storage_phrase = "keep remote on"
+        else:
+            storage_phrase = "send to"
+        f_str = f.storage_object.print_query
+        storage_phrase = f"{phrase} storage"
```

Ruff (0.8.2)

Undefined name `phrase`

(F821)